packages/ember-routing/lib/system/route.js

```diff
@@ -142,14 +142,37 @@ let Route = EmberObject.extend(ActionHandler, Evented, {
   */
 
   /**
+    Sets the name for this route, including a fully resolved name for routes
+    inside engines.
+
+    @private
+    @method setRouteName
+    @param {String} name
+  */
+  setRouteName(name) {
```

Comment:
We should use a symbol or underscore any newly added private methods to make it super obvious that they are private (and avoid naming collisions with userland methods/properties).